src/Services/InventoryStatusService.php

```diff
@@ -0,0 +1,407 @@
+use Wayfair\Repositories\KeyValueRepository;
+
+/**
+ * Service for setting / getting state of Inventory Services
+ */
+class InventoryStatusService
+{
+  const LOG_KEY_STATE_CHECK_FULL = 'fullInventoryStateCheck';
+  const LOG_KEY_STATE_CHANGE_FULL = 'fullInventoryStateChange';
+  const LOG_KEY_RESET_FULL = 'fullInventoryReset';
+  const LOG_KEY_CLEAR_FULL = 'fullInventoryClear';
+
+  const LOG_KEY_STATE_CHECK_PARTIAL = 'partialInventoryStateCheck';
+  const LOG_KEY_STATE_CHANGE_PARTIAL = 'partialInventoryStateChange';
+  const LOG_KEY_RESET_PARTIAL = 'partialInventoryReset';
+  const LOG_KEY_CLEAR_PARTIAL = 'partialInventoryClear';
+
+  const STATUS = 'status';
+  const STATE_CHANGE_TIMESTAMP = 'stateChangeTimestamp';
+  const LAST_COMPLETION = 'lastCompletion';
+  const LAST_ATTEMPT_TIMESTAMP = 'lastAttemptTimestamp';
+  const LAST_ATTEMPT_SUCCEEDED = 'lastAttemptSucceeded';
+
+  const STATE_RUNNING = 'running';
+  const STATE_IDLE = 'idle';
+
+  const INVENTORY_CRON_STATUS_FULL = 'full_inventory_cron_status';
+  const INVENTORY_STATUS_UPDATED_AT_FULL = 'full_inventory_status_updated_at';
+  const INVENTORY_LAST_COMPLETION_FULL = "full_inventory_last_completion";
+  const INVENTORY_LAST_ATTEMPT_FULL = "full_inventory_last_attempt";
+  const INVENTORY_SUCCESS_FULL = "full_inventory_success";
+  const INVENTORY_LAST_SUCCESS_START_FULL = "full_inventory_last_success_start";
+
+  const KEYS_FOR_FULL = [
+    self::INVENTORY_CRON_STATUS_FULL,
+    self::INVENTORY_STATUS_UPDATED_AT_FULL,
+    self::INVENTORY_LAST_COMPLETION_FULL,
+    self::INVENTORY_SUCCESS_FULL,
+    self::INVENTORY_LAST_SUCCESS_START_FULL
+  ];
+
+  const INVENTORY_CRON_STATUS_PARTIAL = 'partial_inventory_cron_status';
+  const INVENTORY_STATUS_UPDATED_AT_PARTIAL = 'partial_inventory_status_updated_at';
+  const INVENTORY_LAST_COMPLETION_PARTIAL = "partial_inventory_last_completion";
+  const INVENTORY_LAST_ATTEMPT_PARTIAL = "partial_inventory_last_attempt";
+  const INVENTORY_SUCCESS_PARTIAL = "partial_inventory_success";
+  const INVENTORY_LAST_SUCCESS_START_PARTIAL = "partial_inventory_last_success_start";
+
+  const KEYS_FOR_PARTIAL = [
+    self::INVENTORY_CRON_STATUS_PARTIAL,
+    self::INVENTORY_STATUS_UPDATED_AT_PARTIAL,
+    self::INVENTORY_LAST_COMPLETION_PARTIAL,
+    self::INVENTORY_LAST_ATTEMPT_PARTIAL,
+    self::INVENTORY_SUCCESS_PARTIAL,
+    self::INVENTORY_LAST_SUCCESS_START_PARTIAL
+  ];
+
+  /**
+   * @var KeyValueRepository
+   */
+  private $keyValueRepository;
+
+  /**
+   * @var LoggerContract
+   */
+  private $logger;
+
+  /**
+   * InventoryStatusService constructor.
+   *
+   * @param KeyValueRepository $keyValueRepository
+   * @param LoggerContract $logger
+   */
+  public function __construct(
+    KeyValueRepository $keyValueRepository,
+    LoggerContract $logger
+  ) {
+    $this->keyValueRepository = $keyValueRepository;
+    $this->logger = $logger;
+  }
+
+  /**
+   * Set the global status of Full Inventory syncing,

```

Comment:
This comment (and others in this file) implies this method is only for full syncing but the parameter determines whether to check full or partial.

```suggestion
   * Set the global status for full or partial inventory syncing,

```

Reply:
yup this was refactored out of a "full only" module - good catch.